src/solver/mlir_common.cpp

```diff
@@ -56,10 +82,13 @@ std::string ConstructBuildOptions(const ConvolutionContext& ctx,
     std::string out_layout = InsertGToLayout(CI::GetOutputLayout(ctx), 'C');
 
     std::string mlir_handle;
-    if (is_xdlops)
-        mlir_handle += std::string(" --x2 ") + "1";
+    if (!ctx.Is2d()) {
+        // Future: Remove this once MLIr supports 3D convolutions
+        return mlir_handle;
+    }
```

Comment:
Can we move this to MLIR?

Reply:
Well, if we wanted to add a series of `--*_depth` options to MLIR and then reject on depth != 0, then we could. Is that a preferable refactor.

Reply:
Can we do an additional check on layout to filter off those that whose length are not 5?

Reply:
On a second look you already have it in the MLIR PR. Please drop this check.

Reply:
... yeah, we do have a rather strict layout filter, and so could filter out 3D that way. Sure.